pkg/tasks/create.go

```diff
@@ -3,25 +3,209 @@ package tasks
+	args := commonArguments(rackName, podName)
+	return CreateClusterTask(ctx, kubeClient, controlapi.CommandCleanup, namespace, cluster, []string{dcName}, args)
+}
+
+// UpgradeSSTables
+
+func CreateUpgradeSSTablesTask(ctx context.Context, kubeClient client.Client, dc *cassdcapi.CassandraDatacenter, rackName string, podName string) (*controlapi.CassandraTask, error) {
+	args := commonArguments(rackName, podName)
+	return CreateTask(ctx, kubeClient, controlapi.CommandUpgradeSSTables, dc, args)
+}
+
+func CreateClusterUpgradeSSTablesTask(ctx context.Context, kubeClient client.Client, namespace, cluster, dcName, rackName, podName string) (*k8ssandrataskapi.K8ssandraTask, error) {
+	args := commonArguments(rackName, podName)
+	return CreateClusterTask(ctx, kubeClient, controlapi.CommandUpgradeSSTables, namespace, cluster, []string{dcName}, args)
+}
+
+// Scrub
+
+func CreateScrubTask(ctx context.Context, kubeClient client.Client, dc *cassdcapi.CassandraDatacenter, rackName string, podName string) (*controlapi.CassandraTask, error) {
+	args := commonArguments(rackName, podName)
+	return CreateTask(ctx, kubeClient, controlapi.CommandScrub, dc, args)
+}
+
+func CreateClusterScrubTask(ctx context.Context, kubeClient client.Client, namespace, cluster, dcName, rackName, podName string) (*k8ssandrataskapi.K8ssandraTask, error) {
+	args := commonArguments(rackName, podName)
+	return CreateClusterTask(ctx, kubeClient, controlapi.CommandScrub, namespace, cluster, []string{dcName}, args)
+}
+
+// Compaction
+
+func CreateCompactionTask(ctx context.Context, kubeClient client.Client, dc *cassdcapi.CassandraDatacenter, rackName string, podName, keyspaceName string, tables []string) (*controlapi.CassandraTask, error) {
+	args, err := compactionArguments(rackName, podName, keyspaceName, tables)
+	if err != nil {
+		return nil, err
+	}
+	return CreateTask(ctx, kubeClient, controlapi.CommandCompaction, dc, args)
+}
+
+func compactionArguments(rackName, podName, keyspaceName string, tables []string) (*controlapi.JobArguments, error) {
+	args := commonArguments(rackName, podName)
+	if keyspaceName != "" {
+		args.KeyspaceName = keyspaceName
+	}
+
+	if keyspaceName == "" && len(tables) > 0 {
+		return nil, fmt.Errorf("keyspace must be specified when tables are specified")
+	}
+
+	if len(tables) > 0 {
+		args.Tables = tables
+	}
+
+	return args, nil
+}
+
+func CreateClusterCompactionTask(ctx context.Context, kubeClient client.Client, namespace, cluster, dcName, rackName, podName, keyspaceName string, tables []string) (*k8ssandrataskapi.K8ssandraTask, error) {
+	args, err := compactionArguments(rackName, podName, keyspaceName, tables)
+	if err != nil {
+		return nil, err
+	}
+	return CreateClusterTask(ctx, kubeClient, controlapi.CommandCompaction, namespace, cluster, []string{dcName}, args)
+}
+
+// Move
+
+// GarbageCollect
```

Comment:
[question] Are we missing the CreateMoveTask functions here?

Reply:
I'm not sure what uses the move functionality at this point, but no - I intentionally omitted that since the original tickets wanted just a subset of tasks to be exposed. 

I think exposing the move might require more knowledge of how it's used (by some automation I guess?)